### PR TITLE
Prepare types for both current api and gRPC

### PR DESCRIPTION
**Pull request by feiskyer**
Prepare types for both current api and gRPC, using types generated by proto.

Part of #278

### CODE REVIEW

**Comment by gnawux** [comment]
gRPC is welcome

File: types/types.proto [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+message ContainerInfo {
+  string name                       = 1;
+  string containerID                = 2;
+  string image                      = 3;
+  string imageID                    = 4;
+  repeated string commands          = 5;
+  repeated string args              = 6;
+  string workingDir                 = 7;
+  repeated ContainerPort ports      = 8;
+  repeated EnvironmentVar env       = 9;
```
Comment (laijs) [inline types/types.proto]: `map<string,string> env` might be a choice

File: types/types.proto [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+
+// TODO: embed Container message to avoid repetition
+message ContainerInfo {
+  string name                       = 1;
+  string containerID                = 2;
+  string image                      = 3;
+  string imageID                    = 4;
+  repeated string commands          = 5;
+  repeated string args              = 6;
+  string workingDir                 = 7;
```
Comment (laijs) [inline types/types.proto]: workdir

File: types/types.proto [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+}
+
+// TODO: embed Container message to avoid repetition
+message ContainerInfo {
+  string name                       = 1;
+  string containerID                = 2;
+  string image                      = 3;
+  string imageID                    = 4;
+  repeated string commands          = 5;
+  repeated string args              = 6;
```
Comment (laijs) [inline types/types.proto]: why it has both `commands` and  `args`

File: types/types.proto [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+    RunningStatus running   = 5;
+    TermStatus terminated   = 6;
+}
+
+// TODO: embed Container message to avoid repetition
+message ContainerInfo {
+  string name                       = 1;
+  string containerID                = 2;
+  string image                      = 3;
+  string imageID                    = 4;
```
Comment (laijs) [inline types/types.proto]: why it has both image and imageID

File: types/types.proto [hunk 5]
```diff
@@ -0,0 +1,127 @@
...
+
+message EnvironmentVar {
+    string env    = 1;
+    string value  = 2;
+}
+
+message VolumeMount {
+    string name       = 1;
+    string mountPath  = 2;
+    bool readOnly     = 3;
```
Comment (laijs) [inline types/types.proto]: use  `repeated string option` instead
the userpod.json will also be updated.

File: types/types.proto (hunk 3 above)
Comment (feiskyer) [inline types/types.proto]: Same meaning with entrypoint and commands

File: types/types.proto [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+  repeated string args              = 6;
+  string workingDir                 = 7;
+  repeated ContainerPort ports      = 8;
+  repeated EnvironmentVar env       = 9;
+  repeated VolumeMount volumeMounts = 10;
+  bool tty                          = 11;
+  string imagePullPolicy            = 12;
+	string podID                      = 13;
+	ContainerStatus status            = 14;
+}
```
Comment (laijs) [inline types/types.proto]: repeated processinfo processes

File: types/types.proto (hunk 5 above)
Comment (feiskyer) [inline types/types.proto]: why `repeated string option`? A struct is more clear than repeated string.

Comment (laijs) [inline types/types.proto]: options a complicated, A struct is overkill.
see also https://github.com/opencontainers/runtime-spec/blob/master/specs-go/config.go#L93-L94

Comment (feiskyer) [inline types/types.proto]: Thanks. Another ref https://github.com/opencontainers/runc/blob/master/libcontainer/specconv/spec_linux.go#L583-L608

File: types/types.proto (hunk 4 above)
Comment (feiskyer) [inline types/types.proto]: image is actually the name of image

File: types/types.proto (hunk 6 above)
Comment (feiskyer) [inline types/types.proto]: Where is processinfo defined?

**Pull request merged**